vmbindings/dummyvm/src/lib.rs

```diff
@@ -26,14 +27,24 @@ impl VMBinding for DummyVM {
     type VMActivePlan = active_plan::VMActivePlan;
     type VMReferenceGlue = reference_glue::VMReferenceGlue;
 
-    /// Allowed maximum alignment as shift by min alignment.    
+    /// Allowed maximum alignment as shift by min alignment.
     const MAX_ALIGNMENT_SHIFT: usize = 6_usize - Self::LOG_MIN_ALIGNMENT as usize;
 
     /// Allowed maximum alignment in bytes.
     const MAX_ALIGNMENT: usize = Self::MIN_ALIGNMENT << Self::MAX_ALIGNMENT_SHIFT;
 }
 
-//#[cfg(feature = "dummyvm")]
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// This is used to ensure we initialize MMTk at a specified timing.
+pub static MMTK_INITIALIZED: AtomicBool = AtomicBool::new(false);
+
 lazy_static! {
-    pub static ref SINGLETON: MMTK<DummyVM> = MMTK::new();
+    pub static ref BUILDER: MMTKBuilder = MMTKBuilder::new();
+    pub static ref SINGLETON: MMTK<DummyVM> = {
+        debug_assert!(!MMTK_INITIALIZED.load(Ordering::Relaxed));
+        let ret = mmtk::memory_manager::gc_init(&BUILDER);
+        MMTK_INITIALIZED.store(true, std::sync::atomic::Ordering::Relaxed);
```

Comment:
If this is only for assertion purpose, `Relaxed` should be fine.

But if we use it for actual inter-thread synchronisation (like in the OpenJDK binding), this store should have at least `Release` order, and loads of `MMTK_INITIALIZED` should be at least `Acquired`.  Or we can use `SeqCst` for both load and store for even stronger order.

I suggest we either add a comment here and mention `Relaxed` is not enough for serious inter-thread synchronisation.  Alternatively, we can use `Acquire` and `Release` (or `SeqCst` for both) here just in case someone would just copy this part of the code into their binding.

Reply:
I changed the ordering to `SeqCst`.